scripts/nginx/init-nginx.sh

```diff
@@ -0,0 +1,722 @@
+
+case "${1:-}" in
+    "--undo")
+        OPERATION="undo"
+        ;;
+    "--add-domain")
+        OPERATION="add"
+        TARGET_DOMAIN="${2:-}"
+        [[ -n "$TARGET_DOMAIN" ]] || error "Domain name required: $0 --add-domain example.com"
+        ;;
+    "--remove-domain")
+        OPERATION="remove"
+        TARGET_DOMAIN="${2:-}"
+        [[ -n "$TARGET_DOMAIN" ]] || error "Domain name required: $0 --remove-domain example.com"
+        ;;
+    "--update-services")
+        OPERATION="update-services"
+        ;;
+    "--list-services")
+        OPERATION="list-services"
+        ;;
+    "--service-status")
+        OPERATION="service-status"
+        ;;
+    "--help"|"-h")
+        echo "SneezyMUD Nginx Setup, Domain and Service Management"
+        echo "Usage:"
+        echo "  sudo $0                          # Initial setup or management menu"
+        echo "  sudo $0 --add-domain DOMAIN      # Add domain to existing setup"
+        echo "  sudo $0 --remove-domain DOMAIN   # Remove domain from setup"
+        echo "  sudo $0 --update-services        # Update nginx config from services.json"
+        echo "  sudo $0 --list-services          # List configured services"
+        echo "  sudo $0 --service-status         # Check service status"
+        echo "  sudo $0 --undo                   # Remove entire setup"
+        echo "  sudo $0 --help                   # Show this help"
+        echo ""
+        echo "Service Management Workflow:"
+        echo "  1. Add/remove service from Docker Compose"
+        echo "  2. Update scripts/nginx/services.json"
+        echo "  3. Run: sudo $0 --update-services"
+        echo ""
+        echo "Backup API Configuration:"
+        echo "  Set BACKUP_API_KEY in .env file to enable /sneezybackups/ endpoint"
+        exit 0
+        ;;
+    "")
+        OPERATION="auto"
+        ;;
+    *)
+        error "Unknown option: $1. Use --help for usage information."
+        ;;
+esac
+
+[[ $EUID -eq 0 ]] || error "Must run with sudo"
+command -v apt-get &> /dev/null || error "Requires Ubuntu/Debian"
+
+# Install jq dependency for JSON processing - avoids requiring manual installation
+if ! command -v jq &> /dev/null; then
+    info "Installing jq for service management"
+    apt-get update -qq
+    apt-get install -y jq
+fi
+
+detect_existing_setup() {
+    # Both domain registry and nginx config must exist for management operations
+    [[ -f "$DOMAINS_FILE" ]] && [[ -f "/etc/nginx/sites-available/sneezy-nginx" ]]
+}
+
+load_domains() {
+    if [[ -f "$DOMAINS_FILE" ]]; then
+        cat "$DOMAINS_FILE"
+    fi
+}
+
+save_domains() {
+    local domains=("$@")
+    # Store one domain per line for easy parsing and management
+    printf "%s\n" "${domains[@]}" > "$DOMAINS_FILE"
+}
+
+validate_domain() {
+    local domain="$1"
+    # Prevent common user input errors that would cause SSL certificate failures
+    [[ "$domain" =~ ^[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$ ]] || error "Invalid domain format: $domain"
```

Comment:
Nit: this regex allows `...com`. At least I think it does.
Try this instead:
```regex
^([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}$
```

Reply:
Yeah, looks like you're right, good catch:

<img width="337" height="179" alt="image" src="https://github.com/user-attachments/assets/48ca7376-e804-41c5-b4f0-4c5559fbf102" />

<img width="395" height="190" alt="image" src="https://github.com/user-attachments/assets/452837e9-85a8-4dc3-8124-027e7eb7d30d" />

Just created https://github.com/sneezymud/sneezymud-docker/issues/20 to remember to fix this, so you don't have to re-approve this PR yet again if I push it now.